### PR TITLE
Basic CRUD Operations docs

### PR DESCRIPTION
**Pull request by KenitoInc**
- [x] Fix text changes in getting started.
- [x] Add more examples.

### CODE REVIEW

File: Odata-docs/client/v7/getting-started.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ The OData Client library allows you to consume data from and interact with OData
 In this article we are going to focus on the latter approach.
 
 ## What are Code Generation Tools
-* A tool used to generate a client proxy file for an Odata service. It used the schema to generate the proxy classes.
+* These are tools used to generate client proxy file for an Odata service. They use the schema to generate the proxy classes.
 * A proxy class shields the client from the complexities involved in invoking the Odata web service.
 * A proxy class contains all the methods and objects exposed by the Odata webservice.
```
Comment (habbes) [inline Odata-docs/client/v7/getting-started.md:25]:
```diff
- webservice
+ web service
```

Comment (KenitoInc) [reply Odata-docs/client/v7/getting-started.md]: #Resolved

File: Odata-docs/client/v7/querying-odata-service.md [hunk 2]
```diff
@@ -12,6 +12,15 @@ ms.topic: article
 
 **Applies To**: [!INCLUDE[appliesto-odataclient](../../includes/appliesto-odataclient-v7.md)]
 
+## Add an entity
+```csharp
+var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
+    var person = Person.CreatePerson("russell", "Russell", "Whyte", new long());
+    context.AddToPeople(person);
+
+    context.SaveChanges();
```
Comment (mumbi-o) [inline Odata-docs/client/v7/querying-odata-service.md:21]: I tried this code and I got this error: https://github.com/smartstore/SmartStoreNET/issues/895. Do we need to capture information about how to solve this?

Comment (KenitoInc) [reply Odata-docs/client/v7/querying-odata-service.md:21]: Are you making an async call?

Comment (mumbi-o) [reply Odata-docs/client/v7/querying-odata-service.md]: Just tried both sync and async and still got the error

Comment (KenitoInc) [reply Odata-docs/client/v7/querying-odata-service.md]: #Resolved

File: Odata-docs/TOC.yml [hunk 3]
```diff
@@ -272,7 +272,7 @@
       - name: Getting Started
         href: /odata/client/getting-started
       - name: Basic CRUD operations
-        href: /odata/client/basic-crud-operations
+        href: /odata/client/querying-odata-service
```
[the remark this answers is not in the document]
Comment (KenitoInc) [reply Odata-docs/TOC.yml]: I reverted the change 
#Resolved

File: Odata-docs/client/v7/getting-started.md [hunk 4]
```diff
@@ -141,5 +141,5 @@ And when you run it, you should see the following output:
 
 As you can see, the OData Connected Service generates C# classes corresponding to your specific OData service and uses the underlying OData client library to serialize and deserialize to and from JSON, saving you a lot of work.
 
-In this tutorial you have learned how to use the OData Connected Service to generate client code to interact with a specific OData service. To learn more about about using the client to read and write to the service, check out the [Basic CRUD operations](./basic-crud-operations.md) tutorial.
+In this tutorial you have learned how to use the OData Connected Service to generate client code to interact with a specific OData service. To learn more about about using the client to read and write to the service, check out the [Basic CRUD Operations](/odata/client/basic-crud-operations.md) tutorial.
```
Comment (gathogojr) [inline Odata-docs/client/v7/getting-started.md:144]: @KenitoInc Having the `.md` on the link I believe is reason for this:
image
You can always click [View] from this table to verify the link open as expected
image

File: Odata-docs/client/v7/using-extensions.md [hunk 5]
```diff
@@ -40,7 +40,7 @@ The IODataClientFactory can be registered by calling the AddODataClient extensio
 services.AddODataClient();
 ```
 
-Once registered, code can accept an IODataClientFactory anywhere services can be injected with dependency injection (DI). The IODataClientFactory can be used to create a generated client proxy instance, then follow [Basic CRUD operations](/odata/client/basic-crud-operations) to use client proxy in business logic:
+Once registered, code can accept an IODataClientFactory anywhere services can be injected with dependency injection (DI). The IODataClientFactory can be used to create a generated client proxy instance, then follow [Basic CRUD Operations](/odata/client/basic-crud-operations.md) to use client proxy in business logic:
```
Comment (gathogojr) [inline Odata-docs/client/v7/using-extensions.md:43]: This [comment](https://github.com/MicrosoftDocs/OData-docs/pull/149/files#r505174949) applies here as well. If you check through other links you'll find that `.md` is left out

File: Odata-docs/client/v7/basic-crud-operations.md [hunk 6]
```diff
@@ -33,6 +42,14 @@ foreach (var person in people)
...
 }
 ```
+or use the `Where()` with `First()` or `Single()` API call:
+
+```c#
+var person = context.People.Where(c => c.UserName == "russellwhyte").First();
+var person = context.People.Where(c => c.UserName == "russellwhyte").Single();
+```
+Note that `First()` will return one object, even if the lambda expression matches multiple objects.
+`Single()` always return one object and will throw an exception if the lambda expression matches multiple objects.
```
Comment (gathogojr) [inline Odata-docs/client/v7/basic-crud-operations.md:52]: I'd expect both `First()` and `Single()` will throw an exception if no person with `UserName` as `russelwhyte` exists?

Comment (KenitoInc) [reply Odata-docs/client/v7/basic-crud-operations.md:52]: Yes. I have added that to the docs.
#Resolved

File: Odata-docs/client/v7/using-extensions.md (hunk 5 above)
Comment (KenitoInc) [reply Odata-docs/client/v7/using-extensions.md]: #Resolved

File: Odata-docs/client/v7/getting-started.md (hunk 4 above)
Comment (KenitoInc) [reply Odata-docs/client/v7/getting-started.md]: #Resolved

File: Odata-docs/client/v7/basic-crud-operations.md [hunk 7]
```diff
@@ -33,6 +42,16 @@ foreach (var person in people)
...
+
+Both `First()` and `Single()` will throw an exception if no person with `UserName` as `russelwhyte` exists?
```
Comment (gathogojr) [inline Odata-docs/client/v7/basic-crud-operations.md:54]:
```suggestion
Both `First()` and `Single()` will throw an exception if no person with `UserName` as `russelwhyte` exists.
```

**Comment by gathogojr** [comment]
As a result changes like the following:
- `/odata/client/using-hooks` => `./using-hooks`
- `/odata/client/code-generation-tool` => `./code-generation-tool`
- `/odata/client/basic-crud-operations` => `./basic-crud-operations`
- `/odata/client/using-httpclient` => `./using-httpclient`,

your PR ended up with quite a number of warnings:
image
The complaint is that file link is invalid. The interesting thing is that the links work so I'm not sure what to make of the warnings. If whatever is being used to validate is not happy, I'd argue that if the first form works - `/odata/client/...`, personally I'd stick with that.
**Or maybe the path needs to be `/odata/client/v7/...` for this form to work in your case**

This summary doesn't look good:
image

**Comment by KenitoInc** [comment]
> As a result changes like the following:
> 
> * `/odata/client/using-hooks` => `./using-hooks`
> * `/odata/client/code-generation-tool` => `./code-generation-tool`
> * `/odata/client/basic-crud-operations` => `./basic-crud-operations`
> * `/odata/client/using-httpclient` => `./using-httpclient`,
> 
> your PR ended up with quite a number of warnings:
> image
> The complaint is that file link is invalid. The interesting thing is that the links work so I'm not sure what to make of the warnings. If whatever is being used to validate is not happy, I'd argue that if the first form works - `/odata/client/...`, personally I'd stick with that.
> **Or maybe the path needs to be `/odata/client/v7/...` for the first form to work in your case**
> 
> This summary doesn't look good:
> image

#resolved

**Review by gathogojr — APPROVED** [review]

**Pull request merged**